src/serverconf/src/main/java/ee/ria/xroad/common/conf/serverconf/IsAuthentication.java

```diff
@@ -90,10 +93,18 @@ public static void verifyClientAuthentication(ClientId client,
                         "Client (%s) has no IS certificates", client);
             }
 
-            if (!isCerts.contains(auth.getCert())) {
-                throw new CodedException(X_SSL_AUTH_FAILED,
-                        "Client (%s) TLS certificate does not match any"
-                                + " IS certificates", client);
+            X509Certificate isCert = isCerts.stream()
+                    .filter(c -> Objects.equals(c, auth.getCert()))
+                    .findFirst()
+                    .orElseThrow(() -> new CodedException(X_SSL_AUTH_FAILED,
+                    "Client (%s) TLS certificate does not match any"
+                                    + " IS certificates", client));
+            try {
+                isCert.checkValidity();
```

Comment:
Since the original `isCerts.contains(auth.getCert())` already verifies that the cert _is_ one of the IS certs, iterating the list is unnecessary. One could after that test safely check the validity of the received certificate (`auth.getCert().checkValidity()`). 
I'd also argue that it would be more readable (and perhaps a bit more efficient) than using a stream.

Reply:
Done